tags.go

```diff
@@ -30,16 +29,62 @@ func serializeTags(name string, tags map[string]string) string {
 		}
 		panic("unreachable")
 	case 2:
-		var a, b tagPair
+		var t0, t1 tagPair
 		for k, v := range tags {
-			b = a
-			a = tagPair{k, replaceChars(v)}
+			t1 = t0
+			t0 = tagPair{k, replaceChars(v)}
 		}
-		if a.dimension > b.dimension {
-			a, b = b, a
+		if t0.dimension > t1.dimension {
+			t0, t1 = t1, t0
 		}
-		return name + prefix + a.dimension + sep + a.value +
-			prefix + b.dimension + sep + b.value
+		return name + prefix + t0.dimension + sep + t0.value +
+			prefix + t1.dimension + sep + t1.value
+	case 3:
+		var t0, t1, t2 tagPair
+		for k, v := range tags {
+			t2 = t1
+			t1 = t0
+			t0 = tagPair{k, replaceChars(v)}
+		}
+		if t1.dimension > t2.dimension {
+			t1, t2 = t2, t1
+		}
+		if t0.dimension > t2.dimension {
+			t0, t2 = t2, t0
+		}
+		if t0.dimension > t1.dimension {
+			t0, t1 = t1, t0
+		}
+		return name + prefix + t0.dimension + sep + t0.value +
+			prefix + t1.dimension + sep + t1.value +
+			prefix + t2.dimension + sep + t2.value
+	case 4:
+		var t0, t1, t2, t3 tagPair
+		for k, v := range tags {
+			t3 = t2
+			t2 = t1
+			t1 = t0
+			t0 = tagPair{k, replaceChars(v)}
+		}
+		if t0.dimension > t1.dimension {
+			t0, t1 = t1, t0
+		}
+		if t2.dimension > t3.dimension {
+			t2, t3 = t3, t2
+		}
+		if t0.dimension > t2.dimension {
+			t0, t2 = t2, t0
+		}
+		if t1.dimension > t3.dimension {
+			t1, t3 = t3, t1
+		}
+		if t1.dimension > t2.dimension {
+			t1, t2 = t2, t1
```

Comment:
I examined the coverage report. this line isn't covered. can you create a test that hits it? Every other line in the file is hit.

![image](https://user-images.githubusercontent.com/734339/54467058-02aecd00-4740-11e9-8d82-17b7cf80265f.png)

Reply:
@btc Iteration order of maps is random (thus 100% coverage cannot be guaranteed) and this network sort was generated, but I added a test that should almost always yield 100% coverage and which yielded 100% coverage on all the runs I made.

Reply:
acknowledged